include/kframework/foundry.md

```diff
@@ -539,6 +552,136 @@ If the `expectRevert()` selector is matched, call the `#setExpectRevert` product
         orBool SELECTOR ==Int selector ( "expectRevert(bytes)" )
 ```
 
+Expecting a specific CALL/CREATE opcode
+---------------------------------------
+
+First we define a sort to identify expected opcodes.
+
+```k
+    syntax OpcodeType ::= ".OpcodeType" | "Call" | "Static" | "Delegate" | "Create" | "Create2"
+```
+
+If the `expect*OPCODE*` selector is matched, the rule will load the account into the state and add the `#setExpectOpcode` production to the K cell to initialize the `<expectedOpcode/>` subconfiguration with the given parameters.
+
+```{.k .bytes}
+    rule [foundry.call.expectStaticCall]:
+         <k> #call_foundry SELECTOR ARGS
+          => #loadAccount #asWord(#range(ARGS, 0, 32))
+          ~> #let CODE_START = 96 #in
+          #let DATA_LENGTH = #asWord(#range(ARGS, 64, 32)) #in
+          #setExpectOpcode #asWord(#range(ARGS, 0, 32)) #range(ARGS, CODE_START, DATA_LENGTH) 0 Static
+          ...
```

Comment:
```suggestion
          ~> #let CODE_START = 96 #in
             #let DATA_LENGTH = #asWord(#range(ARGS, 64, 32)) #in
             #setExpectOpcode #asWord(#range(ARGS, 0, 32)) #range(ARGS, CODE_START, DATA_LENGTH) 0 Static
          ...
```

Reply:
I think it reads a little better formatted like this, up ot you. But do the same throughout the file either way.

Reply:
Also, not sure why we're using the `#let` bindings here. It seems that we're only using the variables defined in them a single time in the `#in` body, and really the point of using a let binding is if the defined variable is used more than once.

I guess you could argue that this reads better with the let-binding. Also up to you, seems correct either way.